lib/SPIRV/SPIRVWriter.h

```diff
@@ -117,10 +127,13 @@ class LLVMToSPIRV : public ModulePass {
   void transFPContract();
   SPIRVValue *transConstant(Value *V);
   SPIRVValue *transValue(Value *V, SPIRVBasicBlock *BB,
-                         bool CreateForward = true);
+                         bool CreateForward = true,
+                         FuncTransMode FuncTrans = FuncTransMode::Decl);
```

Comment:
It feels a bit inelegant to add a new argument to a generic function like `transValue` and `transValueWithoutDecoration` just to control function pointer translation.  I haven't really looked into any alternative ways so maybe it's the only reasonable way; but I'm wondering if you have evaluated different approaches?

Reply:
Well, IMO it's way better than the previous way of function pointers translation. As you may see, this arg allowed us to remove a good amount of code duplication caused by a need to support function pointers in a various instructions, e.g. for Store and InsertElement (and unary ops, etc) we had ~the same funcptrs related code. And as we don't have any globals translation state to keep this option in, I don't see any feasible alternative.

Reply:
Yes I see the removed code duplication which is a good thing.  I was hoping you could deduce the correct behaviour inside transValue itself from the arguments, but I suppose you need the context of the Values involved, so that may not be possible.

Reply:
Yes, basically we need both Use and User here to determine how we want to translate the User. Adding that would result in a major refactoring of all the code using it, doesn't look doable.